core/frontend/src/components/beacon/BeaconTrayMenu.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <v-menu
+    v-if="is_connected_to_wifi"
+    :close-on-content-click="false"
+    nudge-left="150"
+    nudge-bottom="25"
+  >
+    <template
+      #activator="{ on, attrs }"
+    >
+      <v-card
+        elevation="0"
+        color="transparent"
+        v-bind="attrs"
+        v-on="on"
+      >
+        <v-icon
+          v-tooltip="is_connected_to_wifi ?
+            'Connected through a wireless connection, expect degraded performance'
+            :
+            'Connected through a wired connection'"
+          class="px-1 white-shadow"
+          :color="is_connected_to_wifi ? 'yellow' : 'white'"
+        >
+          mdi-ethernet-cable
+        </v-icon>
+      </v-card>
+    </template>
+
+    <v-card
+      elevation="1"
+      width="300"
+    >
+      <v-container>
+        <div class="justify-center">
+          It looks like you are reaching BlueOS via its wi-fi network. This can result in degraded performance.
+          <span
+            v-if="available_wired_domain()"
+            class="mt-3"
+            style="display: block;"
+          >
+            A wired connection is available:
+            <v-btn
+              small
+              :href="`http://${available_wired_domain()}.local`"
+            >
+              Switch to {{ available_wired_domain() }}.local
+            </v-btn>
+          </span>
+        </div>
+      </v-container>
+    </v-card>
+  </v-menu>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+
+import beacon from '@/store/beacon'
+import wifi from '@/store/wifi'
+import { Domain } from '@/types/beacon'
+import { Dictionary } from '@/types/common'
+import back_axios from '@/utils/api'
+
+export default Vue.extend({
+  name: 'BeaconTrayMenu',
+  data() {
+    return {
+      probe_timer: 0,
+      domains: {} as Dictionary<boolean>,
+    }
+  },
+  computed: {
+    wired_interface_domains(): Domain[] {
+      return beacon.available_domains.filter((entry) => entry.service_type === '_http'
+          && (entry.interface.startsWith('eth') || entry.interface.startsWith('enp')))
+    },
+    wireless_interface_domains(): Domain[] {
+      return beacon.available_domains.filter((entry) => entry.service_type === '_http'
+          && (entry.interface.startsWith('wlan0') || entry.interface.startsWith('wlp')))
```

Comment:
And from my search, it appears to be safe to only check for `wl` prefix.
https://github.com/systemd/systemd/blob/main/src/udev/udev-builtin-net_id.c#L1018